app/test_engine/test_collection_discovery.py

```diff
@@ -106,19 +106,17 @@ def discover_test_collections(
     Note that disabled_test_cases and disabled_collections can be used to disable both
     entire collections or individual test cases.
     """
+    disabled_collections = disabled_collections or []
```

Comment:
Does this means `disabled_collections` shouldn't be an Optional?

Reply:
`disabled_collections` has a default value that is read form a file. For the general application run, the disabled test cases will be read from this file. But for unit tests, the method is called without passing this parameter, so the value is None.

Reply:
Not sure I understand. If you don't pass the parameter, the value will be defined by the `disabled_test_collections()` function, that currently may return `None`. Why not make that function return `[]` instead of `None` and get rid of the Optional and of this line?

Reply:
Yeah, you're right. I'll update it